flang/test/Lower/character-assignment.f90

```diff
@@ -89,15 +97,15 @@ subroutine assign_constant(lhs)
   ! CHECK: }
 
   ! Padding
-  ! CHECK-DAG: %[[c32:.*]] = constant 32 : i8
-  ! CHECK-DAG: %[[blank:.*]] = fir.convert %[[c32]] : (i8) -> !fir.char<1>
+  ! CHECK-DAG: %[[blank:.*]] = fir.insert_value %{{.*}}, %c32{{.*}}, %c0{{.*}} : (!fir.char<1>, i8, index) -> !fir.char<1>
   ! CHECK: fir.do_loop %[[j:.*]] = %{{.*}} to %{{.*}} {
-    ! CHECK: %[[jhs_addr:.*]] = fir.coordinate_of %[[lhs]]#0, %[[j]]
+    ! CHECK-DAG: %[[jhs_cast:.*]] = fir.convert %[[lhs]]#0
```

Comment:
What's this cast?

Reply:
The same as the others inside `fir.do_loop` converting from character scalar type to an array of singleton so that `fir.coordinate_of` can be used (we did not need to have it before because we propagated jhs as an array of singleton already). I made it explicit in the test (I already made a few of these explicit, but not all to keep the test easier to maintain).

Reply:
Got it. Thanks.